shadows/framework/src/main/java/org/robolectric/shadows/ShadowWifiManager.java

```diff
@@ -844,20 +845,21 @@ public void networksFoundFromPnoScan(List<ScanResult> scanResults) {
     }
   }
 
-  // Object needs to be used here since PnoScanResultsCallback is hidden. The looseSignatures spec
-  // requires that all args are of type Object.
   @Implementation(minSdk = TIRAMISU)
   @HiddenApi
   protected void setExternalPnoScanRequest(
-      Object ssids, Object frequencies, Object executor, Object callback) {
+      @ClassName("java.util.List<android.net.wifi.WifiSsid>") Object ssids,
+      int[] frequencies,
+      Executor executor,
+      @ClassName("android.net.wifi.WifiManager$PnoScanResultsCallback") Object callback) {
     synchronized (pnoRequestLock) {
       if (callback == null) {
         throw new IllegalArgumentException("callback cannot be null");
       }
 
       List<WifiSsid> pnoSsids = (List<WifiSsid>) ssids;
-      int[] pnoFrequencies = (int[]) frequencies;
-      Executor pnoExecutor = (Executor) executor;
+      int[] pnoFrequencies = frequencies;
+      Executor pnoExecutor = executor;
```

Comment:
Why not use `frequencies` and `executor` directly?

Reply:
done